sympy/functions/elementary/tests/test_exponential.py

```diff
@@ -191,6 +191,13 @@ def test_log_values():
     assert log(0, 2) == zoo
     assert log(0, 5) == zoo
 
+    assert log(sqrt(3)/2 + I*S.Half) == I*pi/6
+    assert log(S.Half - I*sqrt(3)/2) == -I*pi/3
+    assert log(sqrt(2)/2 + I*sqrt(2)/2) == I*pi/4
+    assert log(-S.Half + I*sqrt(3)/2) == I*2*pi/3
```

Comment:
Would it be possible to extend the algorithm so that, for example, `log(-1 + I*sqrt(3))` would give `log(2) + I*2*pi/3`?

Reply:
Yes, it'd definitely be possible to do slightly more advanced simplifications automatically. I'm mainly wondering about the following though and I'd be very happy to read your thoughts on it:

Is such an automatic transformation always wanted? What would be the limits? Hypothetically: `log(-8 + 8*I) = log(2*sqrt(2)) - 3*I*pi/4` or even `log(2*sqrt(6)*I-sqrt(8)) = log(4*sqrt(2)) + 2*I*pi/3`. The presentation of the expressions would change significantly but there'd still be a logarithm involved. (Currently these transformations are left to `expand_complex`.)

Possibly the automatic evaluation could be limited to cases where the log argument is an integer? (Using `as_content_primitive` for factoring out the constant?)

In any case I think I'll also add the necessary processing for recognizing other special angles.

Reply:
It might suffice to consider those arguments that `pure_complex` will resolve, at least to begin with. Then there could be a lookup table for some values of `y/x` that would give the imaginary part of the logarithm.

Reply:
Thanks for your reply. As far as I can tell, `pure_complex` by itself won't be of much help: it bails as soon as a radical is involved.
```
pure_complex(sqrt(2)/2+I*sqrt(2)/2)
pure_complex(sqrt(3)+I)
```
So it'll have to be `re()` and `im()`.
I'll do some more experiments locally before pushing any changes. (I'm marking this PR "WIP" for now.)

Reply:
> (2*sqrt(6)*I-sqrt(8))

FYI: `factor_terms` can be used to get a canonical surd expression:
```python
>>> factor_terms(2*sqrt(6)*I-sqrt(8))
2*(-sqrt(2) + sqrt(6)*I)
>>> factor_terms(2*sqrt(6)*I-sqrt(8), radical=True)
2*sqrt(2)*(-1 + sqrt(3)*I)
```

Reply:
Thanks! Using `factor_terms` as a canonicalization tool for surd expressions is very interesting. I did some quick experiments/research and noticed that `cancel` already calls `factor_terms` as a first step anyway, **but** my current code tries to specifically match `a+b*I` which then fails for the form given by `factor_terms` as output due to the additional multiplicative factor(s). As a result the current PR code evaluates the logarithm just fine for `2*sqrt(6)*I-sqrt(8)` but fails (by design) if the same expression is given in a factored form.

I'll see if I can enlarge the set of properly recognized (and equivalent) expressions without incurring too much of a computational penalty.

Reply:
So I suspect you solved this by removing the multiplicative coefficient here `coeff, arg_ = arg.as_independent(I, as_Add=False)`? i.e. does it solve `2*sqrt(6)*I-sqrt(8)` in factored form?

Reply:
Here's the current output:
```
In [7]: log(2*sqrt(6)*I-sqrt(8))
Out[7]:
            2⋅ⅈ⋅π
log(4⋅√2) + ─────
              3

In [8]: factor_terms(2*sqrt(6)*I-sqrt(8))
Out[8]: 2⋅(-√2 + √6⋅ⅈ)

In [9]: log(_)
Out[9]:
            2⋅ⅈ⋅π
log(4⋅√2) + ─────
              3

In [10]: factor_terms(2*sqrt(6)*I-sqrt(8), radical=True)
Out[10]: 2⋅√2⋅(-1 + √3⋅ⅈ)

In [11]: log(_)
Out[11]:
            2⋅ⅈ⋅π
log(4⋅√2) + ─────
              3

```
So yes, the first `.as_independent` extracts an already present factor and reintegrates it later on in the modulus. I also tried adding a call to `factor_terms` as first step so as to have a simpler `r_ + i_*I` for `cancel(i_/r_)` to take care of, but as far as I could tell there was no discernible benefit (using `timeit`). I also tried to canonicalize using only `factor_terms` and `signsimp` (i.e. without `cancel`), but that didn't work out.